Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/UI/Driver.hs

```diff
@@ -1388,44 +1399,25 @@ getDriverPayments (personId, merchantId_) mbFrom mbTo mbStatus mbLimit mbOffset
           }
       ]
 
-data DriverDuesResp = DriverDuesResp
-  { dues :: [DriverDuesEntity],
-    totalDue :: HighPrecMoney,
-    overdueThreshold :: HighPrecMoney
-  }
-  deriving (Generic, ToJSON, ToSchema, FromJSON)
-
-data DriverDuesEntity = DriverDuesEntity
-  { date :: UTCTime,
-    amount :: Money,
-    earnings :: Money,
-    offers :: [OfferEntity]
-  }
-  deriving (Generic, ToJSON, ToSchema, FromJSON)
-
-data OfferEntity = OfferEntity
-  { title :: Maybe Text,
-    description :: Maybe Text,
-    tnc :: Maybe Text
-  }
-  deriving (Generic, ToJSON, ToSchema, FromJSON)
-
-getDriverDues :: (EsqDBReplicaFlow m r, EsqDBFlow m r, EncFlow m r, CacheFlow m r) => (Id SP.Person, Id DM.Merchant) -> m DriverDuesResp
-getDriverDues (personId, _merchantId) = do
-  driverPlan <- runInReplica $ QDPlan.findByDriverId personId >>= fromMaybeM (NoCurrentPlanForDriver personId.getId)
-  plan <- QPD.findByIdAndPaymentMode driverPlan.planId driverPlan.planType >>= fromMaybeM (PlanNotFound driverPlan.planId.getId)
-  dueInvoices <- runInReplica $ QDF.findAllPendingAndDueDriverFeeByDriverId personId
-  return $
-    DriverDuesResp
-      { dues = buildDriverDuesEntity <$> dueInvoices,
-        totalDue = sum $ map (\dueInvoice -> fromIntegral dueInvoice.govtCharges + fromIntegral dueInvoice.platformFee.fee + dueInvoice.platformFee.cgst + dueInvoice.platformFee.sgst) dueInvoices,
-        overdueThreshold = plan.maxCreditLimit
-      }
+clearDriverDues :: (EsqDBReplicaFlow m r, EsqDBFlow m r, EncFlow m r, CacheFlow m r) => (Id SP.Person, Id DM.Merchant) -> m ClearDuesRes
+clearDriverDues (personId, _merchantId) = do
+  driverPlan <- QDPlan.findByDriverId personId >>= fromMaybeM (NoCurrentPlanForDriver personId.getId)
+  void $ QPD.findByIdAndPaymentMode driverPlan.planId driverPlan.planType >>= fromMaybeM (PlanNotFound driverPlan.planId.getId)
+  dueDriverFees <- QDF.findAllPendingAndDueDriverFeeByDriverId personId
```

Comment:
`findAllPendingAndDueDriverFeeByDriverId` checks `RECURRING_INVOICE only`. It will not show you autopay's invoice. You need to check if an autopay payment is in progress too.

Reply:
Sure will add REI also.